portal-bridge/src/cli.rs

```diff
@@ -154,6 +159,76 @@ pub struct BridgeConfig {
         help = "The maximum number of concurrent offer rpc requests for state bridge."
     )]
     pub offer_limit: usize,
+
+    #[arg(
+        default_value_t = ENR_OFFER_LIMIT,
+        long = "enr-offer-limit",
+        help = "The maximum number of enrs to gossip per content key (STATE BRIDGE ONLY)."
+    )]
+    pub enr_offer_limit: usize,
+
+    #[arg(
+        long = "filter-clients",
+        help = "Filter clients out from offer request (STATE BRIDGE ONLY).",
+        value_parser = client_parser,
+        default_value = "",
+    )]
+    pub filter_clients: Arc<Vec<ClientType>>,
```

Comment:
Well, I'm not sure exactly why the problem is caused, but it seems to come from how clap handles the custom parser method. The compiler doesn't complain, but when you run the program (without using `Arc`), you get the following error.
```
Mismatch between definition and access of `filter_clients`. Could not downcast to portal_bridge::cli::ClientType, need to downcast to alloc::vec::Vec<portal_bridge::cli::ClientType>
```
However, in this case (`ClientType`), since we really don't need any custom parsing logic (like `subnetwork`) we can just use the `value_delimiter` as you pointed out